File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} n
 * @return {ListNode}
 */
var removeNthFromEnd = function(head, n) {
    let temp = head;
    let cnt = 0;
    while (temp) {
        cnt++;
        temp = temp.next;
    }
    let rem = cnt - n;
    temp = new ListNode(0, head);
    const res = temp;
    while (rem) {
        temp = temp.next;
        rem--;
    }
    temp.next = temp.next ? temp.next.next: null;
    return res.next;
};

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:

        temp = head;
        cnt = 0;
        while temp:
            cnt += 1
            temp = temp.next
        rem = cnt - n
        temp = ListNode(0, head)
        res = temp
        while rem:
            temp = temp.next
            rem -= 1
        temp.next = temp.next.next if temp.next else None
        return res.next